src/test/xml_results.go

```diff
@@ -8,172 +8,510 @@ import (
-	Error      *jUnitXMLFailure `xml:"error,omitempty"`
-	Skipped    *jUnitXMLFailure `xml:"skipped,omitempty"`
-	Time       float64          `xml:"time,attr,omitempty"`
-	Type       string           `xml:"type,attr,omitempty"`
-	Success    string           `xml:"success,attr,omitempty"`
-	Stacktrace string           `xml:"stacktrace,attr,omitempty"`
-	Stdout     string           `xml:"stdout,omitempty"`
-	Stderr     string           `xml:"stderr,omitempty"`
-}
-
-func (j jUnitXMLTest) Duration() time.Duration {
-	return time.Duration(j.Time * float64(time.Second))
+	Name string `xml:"name,attr"`
+
+	Assertions uint   `xml:"assertions,attr,omitempty"`
+	ClassName  string `xml:"classname,attr,omitempty"`
+	Status     string `xml:"status,attr,omitempty"`
+	Timed             `xml:"time,attr,omitempty"`
+
+	Error        *jUnitXMLError         `xml:"error,omitempty"`
+	FlakyError   []jUnitXMLFlaky        `xml:"flakyError,omitempty"`
+	RerunError   []jUnitXMLRerunError   `xml:"rerunError,omitempty"`
+	Failure      *jUnitXMLFailure       `xml:"failure,omitempty"`
+	FlakyFailure []jUnitXMLFlaky        `xml:"flakyFailure,omitempty"`
+	RerunFailure []jUnitXMLRerunFailure `xml:"rerunFailure,omitempty"`
+	Skipped      *jUnitXMLSkipped       `xml:"skipped,omitempty"`
+	Stdout       string                 `xml:"system-out,omitempty"`
+	Stderr       string                 `xml:"system-err,omitempty"`
+}
+
+type jUnitXMLProperties struct {
+	Property []jUnitXMLProperty `xml:"property"`
+}
+
+type jUnitXMLProperty struct {
+	Name  string `xml:"name,attr"`
+	Value string `xml:"value,attr"`
+}
+
+type jUnitXMLError struct {
+	Message string `xml:"message,attr,omitempty"`
+	Type    string `xml:"type,attr"`
+
+	Traceback string `xml:",chardata"`
 }
 
 type jUnitXMLFailure struct {
-	Message   string `xml:"message,attr,omitempty"`
-	Type      string `xml:"type,attr,omitempty"`
+	Message string `xml:"message,attr,omitempty"`
+	Type    string `xml:"type,attr"`
+
+	Traceback string `xml:",chardata"`
+}
+
+type jUnitXMLFlaky struct {
+	Message string `xml:"message,attr,omitempty"`
+	Type    string `xml:"type,attr"`
+
+	Traceback string `xml:",chardata"`
+	Stdout    string `xml:"system-out,omitempty"`
+	Stderr    string `xml:"system-err,omitempty"`
+}
+
+type jUnitXMLRerunError struct {
+	Message string `xml:"message,attr,omitempty"`
+	Type    string `xml:"type,attr"`
+
 	Traceback string `xml:",chardata"`
+	Stdout    string `xml:"system-out,omitempty"`
+	Stderr    string `xml:"system-err,omitempty"`
+}
+
+type jUnitXMLRerunFailure struct {
+	Message string `xml:"message,attr,omitempty"`
+	Timed          `xml:"time,attr"`
+	Type    string `xml:"type,attr"`
+
+	Traceback string `xml:",chardata"`
+	Stdout    string `xml:"system-out,omitempty"`
+	Stderr    string `xml:"system-err,omitempty"`
+}
+
+type jUnitXMLSkipped struct {
+	Message string `xml:"message,attr,omitempty"`
+}
+
+type Timed struct {
+	Time float64 `xml:"time,attr"`
```

Comment:
I wonder if this should really be a `time.Duration` and handle unmarshalling instead of converting later?